src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,48 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final Random RANDOM = new Random();
+    private static final ColorSupplier COLOR_SUPPLIER = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        Type figures = Type.values()[new Random().nextInt(Type.values().length)];

```

Comment:
Think about which variables should be local in the method and which should be class-level. You have random variable, but you use `new Random()`